### PR TITLE
Add test flag

### LINKED ISSUE #24: Tests
Just found out about this and now I am intrigued :D

Would it be possible to add running `passthru.tests` for a derivation being updated before committing?

Example:

`nix-update --commit --test sbt` and the commit step will only run if the tests have passed.

### PR DESCRIPTION
**Pull request by NeQuissimus**
Closes #24 

This is sort of what I had in mind to add testing.
Might need some `is_nix_flakes()` love, I am not super familiar with that.

### CODE REVIEW

File: nix_update/__init__.py [hunk 1]
```diff
@@ -132,14 +136,29 @@ def nix_build(options: Options) -> None:
...
 
+def nix_test(git_dir: str, package: Package, system: Optional[str]) -> None:
+    if not package.tests or not system:
+        die(f"Package {package.name} does not define any tests for {system}")
+    else:
+        tests = []
+        for t in package.tests:
+            tests.append("-A")
+            tests.append(f"tests.{t}.{system}")
+        cmd = ["nix-build", f"{git_dir}/nixos/release.nix"] + tests
```
Comment (Mic92) [inline nix_update/__init__.py:147]: Not all tests are nixos tests. Sometimes people just reference derivations here. I will have a look at that.

**Comment by Mic92** [comment]
I switched now to nixFlakes by default to keep the test matrix small.

File: nix_update/__init__.py (hunk 1 above)
Comment (NeQuissimus) [reply nix_update/__init__.py:147]: Interesting, I don't think I have ever seen that. I basically tried to do something similar to @ofborg here

**Comment by NeQuissimus** [comment]
Rebased on top of `master`

File: nix_update/__init__.py (hunk 1 above)
Comment (Mic92) [reply nix_update/__init__.py:150]: It's an edge case I have seen it only a handful times.

**Comment by Mic92** [comment]
It's now a bit shorter and works with other package set that are not nixpkgs.

File: nix_update/__init__.py [hunk 2]
```diff
@@ -139,29 +139,26 @@ def nix_build(options: Options) -> None:
...
-        cmd = ["nix-build", f"{git_dir}/nixos/release.nix"] + tests
-        run(cmd)
+def nix_test(package: Package) -> None:
+    if not package.tests:
+        die(f"Package '{package.name}' does not define any tests")
+
+    tests = []
+    for t in package.tests:
+        tests.append("-A")
+        tests.append(f"{package.attribute}.tests.{t}")
```
Comment (NeQuissimus) [inline nix_update/__init__.py:149]: Ah yeah, that's clever. Much better :)

**Comment by NeQuissimus** [comment]
@Mic92 Any chance you could tag this, so we can update the version in nixpkgs? :)

**Comment by Mic92** [comment]
https://github.com/Mic92/nix-update/releases/tag/0.2

**Comment by Mic92** [comment]
Remember to also use nixFlakes in nixpkgs when updating. Older nix versions no longer work.

**Pull request merged**